uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/manager-leave/components/manager-leave-calendar_test.js

```diff
@@ -1,30 +1,412 @@
+        });
+      });
+
+      describe('period label', function () {
+        var label, period;
+
+        describe('when the period is current', function () {
+          beforeEach(function () {
+            period = _(controller.absencePeriods).find(function (period) {
+              return period.current;
+            });
+            label = controller.labelPeriod(period);
+          });
+
+          it('adds Current Period to the label', function () {
+            expect(label).toBe('Current Period (' + period.title + ')');
+          });
+        });
+
+        describe('when the period is not current', function () {
+          beforeEach(function () {
+            period = _(controller.absencePeriods).filter(function (period) {
+              return !period.current;
+            }).sample();
+            label = controller.labelPeriod(period);
+          });
+
+          it('returns the title as it is', function () {
+            expect(label).toBe(period.title);
+          });
+        });
+      });
+
+      describe('getDayName', function () {
+        var anySunday;
+
+        beforeEach(function () {
+          anySunday = '2017/02/05';
+        });
+
+        it('returns day name of a date', function () {
+          expect(controller.getDayName(anySunday)).toBe('Sun');
+        });
+      });
+
+      describe('getMonthData', function () {
+        var januaryMonth = 0,
+          returnValue;
+
+        beforeEach(function () {
+          returnValue = controller.getMonthData(1, januaryMonth);
+        });
+
+        it('returns the date which are from the month of january', function () {
+          _.each(returnValue, function (dateObject) {
+            expect(moment(dateObject.date).month()).toBe(januaryMonth);
+          });
+        });
+      });
+
+      describe('filterContacts', function() {
+        beforeEach(function() {
+          controller.filteredContacts = ContactAPIMock.mockedContacts().list;
+        });
+
+        describe('when contacts with leaves filter is false', function() {
+          var returnValue;
+
+          beforeEach(function() {
+            controller.filters.contacts_with_leaves = false;
+            returnValue = controller.filterContacts();
+          });
+
+          it('does not filter the contacts', function() {
+            expect(returnValue).toEqual(controller.filteredContacts);
+          });
+        });
+
+        describe('when contacts with leaves filter is true', function() {
+          var returnValue;
+
+          beforeEach(function() {
+            controller.filters.contacts_with_leaves = true;
+            returnValue = controller.filterContacts();
+          });
+
+          it('filters the contacts', function() {
+            var filteredContacts = controller.filteredContacts.filter(function (contact) {
+              return leaveRequestData.all().values.find(function (request) {
+                return request.contact_id == contact.id;
+              });
```

Comment:
is above the same logic as `controller.filterContacts()` ? what happens if logic in controller changes?